Sources/SWBTestSupport/CoreBasedTests.swift

```diff
@@ -272,53 +272,55 @@ extension CoreBasedTests {
             return nil
         }
     }
-    package var linkPath: Path? {
+    package var lldPath: Path? {
         get async throws {
             let (core, defaultToolchain) = try await coreAndToolchain()
-            if core.hostOperatingSystem != .windows {
-                // Most unixes have a link executable, but that is not a linker
-                return nil
-            }
-            if let executable = defaultToolchain.executableSearchPaths.findExecutable(operatingSystem: core.hostOperatingSystem, basename: "link") {
+            if let executable = defaultToolchain.executableSearchPaths.findExecutable(operatingSystem: core.hostOperatingSystem, basename: "ld.lld") {
                 return executable
             }
             for platform in core.platformRegistry.platforms {
-                if let executable = platform.executableSearchPaths.findExecutable(operatingSystem: core.hostOperatingSystem, basename: "link") {
+                if let executable = platform.executableSearchPaths.findExecutable(operatingSystem: core.hostOperatingSystem, basename: "ld.lld") {
                     return executable
                 }
             }
             return nil
         }
     }
-
-    package var lldPath: Path? {
+    package var goldPath: Path? {
         get async throws {
             let (core, defaultToolchain) = try await coreAndToolchain()
-            if let executable = defaultToolchain.executableSearchPaths.findExecutable(operatingSystem: core.hostOperatingSystem, basename: "ld.ld") {
+            if let executable = defaultToolchain.executableSearchPaths.findExecutable(operatingSystem: core.hostOperatingSystem, basename: "ld.gold") {
                 return executable
             }
             for platform in core.platformRegistry.platforms {
-                if let executable = platform.executableSearchPaths.findExecutable(operatingSystem: core.hostOperatingSystem, basename: "ld.ld") {
+                if let executable = platform.executableSearchPaths.findExecutable(operatingSystem: core.hostOperatingSystem, basename: "ld.gold") {
                     return executable
                 }
             }
             return nil
         }
     }
+    package func linkPath(_ targetArchitecture: String) async throws -> Path? {
+        let (core, defaultToolchain) = try await self.coreAndToolchain()
+        let prefixMapping = [ "x86_64": "x64", "aarch64": "arm64", "i386": "x86", "thumbv7" : "arm" ]
```

Comment:
Sorry for the churn but I ended up changing thumbv7 to armv7 in #304. And i386 should be i686.